.github/workflows/test-run.yml

```
@@ -1,36 +1,60 @@
-# This is a basic workflow to help you get started with Actions
+# A basic CI workflow that tests if the app works and then deploys it on an openstack machine
 
 name: CI
 
-# Controls when the workflow will run
 on:
-  # Triggers the workflow on push or pull request events but only for the master branch
   push:
     branches: [master]
   pull_request:
     branches: [master]
 
-  # Allows you to run this workflow manually from the Actions tab
   workflow_dispatch:
 
-# A workflow run is made up of one or more jobs that can run sequentially or in parallel
 jobs:
-  # This workflow contains a single job called "build"
   build-and-test:
-    # The type of runner that the job will run on
     runs-on: ubuntu-latest
 
-    # Steps represent a sequence of tasks that will be executed as part of the job
     steps:
-      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      # Get repository
       - uses: actions/checkout@v2
-      - name: setup app
+      - name: Checkout
+
+        #install requirements.txt
         uses: actions/setup-python@v2
         with:
           python-version: "3.8"
       - name: Instal dependencies
         run: |
           python3 -m pip install --upgrade pip
           pip3 install -r requirements.txt
+
+      #run pytest unit tests
       - name: Run main tests
         run: pytest
+
+  deploy:
+    runs-on: ubuntu-latest
+    needs: [build-and-test]
+    steps:
+      #get the repo
+      - uses: actions/checkout@v2
+      - name: Checkout
+
+      #send it to the openstack machine via scp
+      - name: Copy files to machine
+        # if: github.ref == 'refs/heads/master'
+        run: |
+          scp -oProxyJump="ubuntu@149.156.182.181" -i ${{ secrets.DEPLOY_C3_TEST }} -r ..\yaptide ubuntu@30.30.30.8:~
+
+      - name: Build and run the app via ssh
+        # if: github.ref == 'refs/heads/master'
+      - uses: appleboy/ssh-action@master
```

```suggestion
      uses: appleboy/ssh-action@master
```

I suspect this should belong to a step with name above

---

I misunderstood the puprose of the "-", there are a few simmilar errors.

Fixed along with the other errors in b66db34d23825b5daf619e586ef1f6ba1cc16003, also introduced a more consistent step structure.